File: leetcode-py/leetcode_py/palindrome_number.py


# Problem 9.
# https://leetcode.com/problems/palindrome-number/
class Solution:
    def is_palindrome(self, x: int) -> bool:
        x = str(x)

        while len(x) >= 2:
            x, first = x[1:], x[0]
            x, last = x[:-1], x[-1]

            if first != last:
                return False

        return True

File: leetcode-rs/src/solutions/palindrome_number.rs
// Problem 9.
// https://leetcode.com/problems/palindrome-number/
#[allow(dead_code)]
pub fn is_palindrome(x: i32) -> bool {
    if x < 0 || (x > 0 && x % 10 == 0) {
        return false;
    }

    x.to_string().chars().rev().eq(x.to_string().chars())
}

#[cfg(test)]
mod tests {
    use super::*;
    use rstest::rstest;

    #[rstest]
    #[case(121, true)]
    #[case(-121, false)]
    #[case(10, false)]
    fn is_palindrome_test(#[case] input: i32, #[case] expected: bool) {
        assert_eq!(is_palindrome(input), expected);
    }
}
